src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/CloudProvider.java

```diff
@@ -58,4 +65,22 @@
     public String addToInstancePublicIp(Infrastructure infrastructure, String instanceId, String desiredIp);
 
     public void removeInstancePublicIp(Infrastructure infrastructure, String instanceId, String desiredIp);
+
+    /**
+     * Collect tags and ensure that mandatory connector-iaas tag key is not duplicated
+     *
+     * @param connectorIaasTag  mandatory connector-iaas tag
+     * @param options           instance's options that may contain tags
+     * @return  the list of all tags
+     */
+    public default List<Tag> retrieveAllTags(Tag connectorIaasTag, Options options) {
```

Comment:
Can you move this method in a separate Spring Component, for example TagManager? You can directly inject the component after in all the cloud providers without the need to pass the  Tag connectorIaasTag parameter when creating the instances. Also the property @Value("${connector-iaas-tag.key:proactive-connector-iaas}") can be directly declared in this new TagsManager component